boards/espressif/esp32c6_devkitc/esp32c6_devkitc.yaml

```diff
@@ -0,0 +1,14 @@
+identifier: esp32c6_devkitc
+name: ESP32-C6
+vendor: espressif
+type: mcu
+arch: riscv
+toolchain:
+  - zephyr
+supported:
+  - gpio
```

Comment:
If you think there are additional peripherals already supported, add it here. Consider checking additional overlays in samples/test folder as well if needed.